tests/clients/test_base.py

```diff
@@ -184,6 +207,21 @@ def test_delete(requests_mock, session, patch_auth):
     assert response == DummyObject(foo="bar")
 
 
+def test_stream(mocker):
+    response_content = mocker.Mock()
+    response_content.iter_lines.return_value = STREAM_RESPONSE
+    mocker.patch.object(
+        BaseClient,
+        "_get_raw",
+        return_value=response_content,
+    )
+
+    client = BaseClient(mocker.Mock(), mocker.Mock())
+    with client._stream("endpoint") as sse_stream:
+        for client_sse, mock_sse in zip(sse_stream, SSE_MOCK_MESSAGES):
+            assert client_sse == mock_sse
```

Comment:
`zip` happily accepts sequences of different lengths, and stops iterating when the shortest sequence it is passed ends. In this case, if `sse_stream` returned more entries than specified by `SSE_MOCK_MESSAGES`, this test would not catch it. I think it would be a bit safer to instead do:

```suggestion
    with client._stream("endpoint") as sse_stream:
        messages = list(sse_stream)
    assert messages == SSE_MOCK_MESSAGES
```